### PR TITLE
Add a cost to building towers and make enemies drop some loot

### PR DESCRIPTION
**Pull request by mystal**
Really simple for now. We should extend this to make all the values tunable, but probably based on whatever design we want. i.e., if we want drops to be based on enemy type or level or whatever. Same with tower costs.

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -25,7 +28,10 @@ struct State<'a, 'b> {
 
 impl<'a, 'b> ggez::event::EventHandler for State<'a, 'b> {
     fn mouse_button_down_event(&mut self, _ctx: &mut Context, button: MouseButton, x: f32, y: f32) {
-        if button == MouseButton::Left && *self.world.read_resource::<PlayState>() == PlayState::Play {
+        // TODO: Move this to a system.
+        // If the player clicks on an open spot on the grid and has enough bits, then build a tower.
+        if button == MouseButton::Left && *self.world.read_resource::<PlayState>() == PlayState::Play &&
```
Comment (Ablexxive) [inline src/main.rs]: Can we put each `&&` on a separate line?

File: src/main.rs [hunk 2]
```diff
@@ -50,6 +56,11 @@ impl<'a, 'b> ggez::event::EventHandler for State<'a, 'b> {
                     .with(Faction::Player)
                     .with(Shooter { seconds_per_attack: 1.0, cooldown: 0.0, attack_radius: 100.0 })
                     .build();
+
+                // Spend resources!
+                // TODO: Make the cost tunable in data somehow.
```
Comment (Ablexxive) [inline src/main.rs:62]: Agreed! Maybe a "Cost" component that can be initialized on creation. That will let us do silly things like make a monster tower that will spawn monsters when clicked but only at a cost.

**Review by Ablexxive — APPROVED** [review]

File: src/main.rs (hunk 1 above)
Comment (mystal) [reply src/main.rs]: Done!

File: src/main.rs (hunk 2 above)
Comment (mystal) [reply src/main.rs:62]: Yeah, I think a component could work well :) Let's have a design discussion!

**Pull request merged**